CliClient/tests/feature_NoteHistory.js

```diff
@@ -33,106 +33,147 @@ describe('integration_ForwardBackwardNoteHistory', function() {
 		const notes1 = await createNTestNotes(5, folders[1]);
-		await time.msleep(100);
+		await testApp.wait();
 
 		testApp.dispatch({ type: 'FOLDER_SELECT', id: id(folders[0]) });
-		await time.msleep(100);
+		await testApp.wait();
 
 		let state = testApp.store().getState();
 		expect(state.backwardHistoryNotes).toEqual([]);
 		expect(state.forwardHistoryNotes).toEqual([]);
 
 		testApp.dispatch({ type: 'FOLDER_SELECT', id: id(folders[1]), historyAction: 'goto' });
-		await time.msleep(100);
+		await testApp.wait();
 
 		state = testApp.store().getState();
 		expect(ids(state.backwardHistoryNotes)).toEqual(ids([notes0[4]])); // notes0[4] was last created
 		expect(ids(state.forwardHistoryNotes)).toEqual([]);
 
 		testApp.dispatch({ type: 'FOLDER_SELECT', id: id(folders[0]), historyAction: 'goto' });
-		await time.msleep(100);
+		await testApp.wait();
 
 		state = testApp.store().getState();
 		expect(ids(state.backwardHistoryNotes)).toEqual(ids([notes0[4], notes1[4]]));
 		expect(state.forwardHistoryNotes).toEqual([]);
 
 		goBackWard(state);
-		await time.msleep(100);
+		await testApp.wait();
 
 		state = testApp.store().getState();
 		expect(ids(state.backwardHistoryNotes)).toEqual(ids([notes0[4]]));
 		expect(ids(state.forwardHistoryNotes)).toEqual(ids([notes0[4]]));
 
 		goForward(state);
-		await time.msleep(100);
+		await testApp.wait();
 
 		state = testApp.store().getState();
 		expect(ids(state.backwardHistoryNotes)).toEqual(ids([notes0[4], notes1[4]]));
 		expect(state.forwardHistoryNotes).toEqual([]);
+	}));
 
 
-	}));
+	it('should save history when searching for a note', asyncTest(async () => {
+		const folders = await createNTestFolders(2);
+		await testApp.wait();
+		const notes0 = await createNTestNotes(5, folders[0]);
+		const notes1 = await createNTestNotes(5, folders[1]);
+		await testApp.wait();
 
+		testApp.dispatch({ type: 'FOLDER_SELECT', id: id(folders[0]) });
+		await testApp.wait();
+
+		let state = testApp.store().getState();
+		expect(state.backwardHistoryNotes).toEqual([]);
+		expect(state.forwardHistoryNotes).toEqual([]);
+
+		testApp.dispatch({ type: 'FOLDER_SELECT', id: id(folders[1]), historyAction: 'goto' });
+		await testApp.wait();
+
+		state = testApp.store().getState();
+		expect(ids(state.backwardHistoryNotes)).toEqual(ids([notes0[4]])); // notes0[4] was last created
+		expect(ids(state.forwardHistoryNotes)).toEqual([]);
+
+		const searchId = uuid.create();
+		testApp.dispatch({
+			type: 'SEARCH_UPDATE',
+			search: {
+				id: searchId,
+				title: notes0[0].title,
+				query_pattern: notes0[0].title,
+				query_folder_id: null,
+				type: BaseModel.TYPE_SEARCH,
+			},
+		});
+		await testApp.wait();
+
+		testApp.dispatch({
+			type: 'SEARCH_SELECT',
+			id: searchId,
+		});
+		await testApp.wait();
+
+		state = testApp.store().getState();
+		expect(ids(state.backwardHistoryNotes)).toEqual(ids([notes0[4], notes1[4]]));
+		expect(ids(state.forwardHistoryNotes)).toEqual([]);
+	}));
 });
```

Comment:
@naviji  we also need to check it works with conflict notes. These are notes that initially appear in a special folder, not their parent.  I suspect forward/backward work OK with them, however they are so unusual we need to test with them.

Reply:
I don't understand what conflict notes are yet. Could you give an example? Also, how should the arrows behave with these conflict notes?